thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyFunctionResource.java

```diff
@@ -1,57 +1,145 @@
 import javax.ws.rs.core.Response;
 
 import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.joda.time.DateTime;
+import org.quartz.CronExpression;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static com.linkedin.thirdeye.anomaly.detection.lib.AutotuneMethodType.*;
 
-@Path("thirdeye/function")
+
+@Path("dashboard/anomaly-function")
 @Produces(MediaType.APPLICATION_JSON)
 public class AnomalyFunctionResource {
 
   private static final Logger LOG = LoggerFactory.getLogger(AnomalyFunctionResource.class);
 
+  private static final ThirdEyeCacheRegistry CACHE_REGISTRY_INSTANCE = ThirdEyeCacheRegistry.getInstance();
+  private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
+  private static ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
   private final Map<String, Object> anomalyFunctionMetadata = new HashMap<>();
   private final AnomalyFunctionFactory anomalyFunctionFactory;
+  private AnomalyFunctionManager anomalyFunctionDAO;
+  private MergedAnomalyResultManager mergedAnomalyResultDAO;
+  private RawAnomalyResultManager rawAnomalyResultDAO;
+  private AutotuneConfigManager autotuneConfigDAO;
+  private EmailConfigurationManager emailConfigurationDAO;
+
+  private static final String DEFAULT_CRON = "0 0 0 * * ?";
+  private static final String UTF8 = "UTF-8";
+  private static final String DEFAULT_FUNCTION_TYPE = "WEEK_OVER_WEEK_RULE";
 
   public AnomalyFunctionResource(String functionConfigPath) {
     buildFunctionMetadata(functionConfigPath);
+    this.anomalyFunctionDAO = DAO_REGISTRY.getAnomalyFunctionDAO();
+    this.rawAnomalyResultDAO = DAO_REGISTRY.getRawAnomalyResultDAO();
+    this.mergedAnomalyResultDAO = DAO_REGISTRY.getMergedAnomalyResultDAO();
+    this.autotuneConfigDAO = DAO_REGISTRY.getAutotuneConfigDAO();
+    this.emailConfigurationDAO = DAO_REGISTRY.getEmailConfigurationDAO();
     this.anomalyFunctionFactory = new AnomalyFunctionFactory(functionConfigPath);
   }
 
+  // Partially update anomaly function
+
+  /**
+   * Update the properties of the given anomaly function id
+   * @param id
+   *    The id of the given anomaly function
+   * @param config
+   *    The json string defining the function properties to be updated, ex. {"pValueThreshold":"0.01",...}
+   * @return
+   *    OK if the properties are successfully updated
+   */
+  @PUT
+  @Path("/update")
+  public Response updateAnomalyFunctionProperties (@QueryParam("id") @NotNull Long id,
+      @QueryParam("config") @NotNull String config) {
+    if(id == null || anomalyFunctionDAO.findById(id) == null) {
+      String msg = "Unable to update function properties. " + id + " doesn't exist";
+      LOG.warn(msg);
+      return Response.status(Response.Status.BAD_REQUEST).entity(msg).build();
+    }
+
+    if(StringUtils.isNotBlank(config)) {
+      Map<String, String> configs = Collections.emptyMap();
+      try {
+        configs = OBJECT_MAPPER.readValue(config, Map.class);
```

Comment:
use proper request object instead of passing string and then converting that to an object.